app/code/community/Inchoo/SocialConnect/Block/Linkedin/Button.php

```diff
@@ -35,16 +40,24 @@ protected function _construct() {
 
         $this->userInfo = Mage::registry('inchoo_socialconnect_linkedin_userinfo');
 
-        // CSRF protection
-        Mage::getSingleton('core/session')->setLinkedinCsrf($csrf = md5(uniqid(rand(), true)));
-        $this->client->setState($csrf);
-
         Mage::getSingleton('customer/session')
             ->setSocialConnectRedirect(Mage::helper('core/url')->getCurrentUrl());
 
         $this->setTemplate('inchoo/socialconnect/linkedin/button.phtml');
     }
 
+    protected function _beforeToHtml()
+    {
+        if (!static::$csrf) {
+            static::$csrf = Mage::getSingleton('core/session')->getLinkedinCsrf();
```

Comment:
Err... `linkedin_csrf` is not set on the session yet. Shouldn't that be `md5(uniqid(rand(), true))`?